db/multi_cf_iterator.cc

```diff
@@ -9,54 +9,95 @@
 
 namespace ROCKSDB_NAMESPACE {
 
-void MultiCfIterator::SeekToFirst() {
+void MultiCfIterator::SeekCommon(
+    const std::function<void(Iterator*)>& child_seek_func,
+    Direction direction) {
+  direction_ = direction;
   Reset();
   int i = 0;
   for (auto& cfh_iter_pair : cfh_iter_pairs_) {
     auto& cfh = cfh_iter_pair.first;
     auto& iter = cfh_iter_pair.second;
-    iter->SeekToFirst();
+    child_seek_func(iter.get());
     if (iter->Valid()) {
       assert(iter->status().ok());
-      min_heap_.push(MultiCfIteratorInfo{iter.get(), cfh, i});
+      if (direction_ == kReverse) {
+        auto& max_heap = std::get<MultiCfMaxHeap>(heap_);
```

Comment:
Maybe I should get the min/max heap before the loop... (and use `get_if` since `get` can throw if attempting to get the wrong type)

Reply:
Yes, that would be a bit more efficient. In general, `direction_` seems redundant since `std::variant` can tell which variant it holds at any given moment (see also the `index()` method)